src/renderer/library/redux/sagas/opds.ts

```diff
@@ -28,6 +28,34 @@ export const SEARCH_TERM = "{searchTerms}";
 // Logger
 const debug = debug_("readium-desktop:renderer:redux:saga:opds");
 
+function* resetSearchUrl(rootFeedIdentifier: string) {
+
+    try {
+
+        const previousLocationPathname = yield* selectTyped(
+            (state: ILibraryRootState) => state.history);
+
+        const { rootFeedIdentifier: previousRootFeedIdentifier } =
+            parseOpdsBrowserRoute(previousLocationPathname[previousLocationPathname.length - 1]?.pathname) || {};
+
+
+        debug("RESET search url: ", previousRootFeedIdentifier, " != ", rootFeedIdentifier);
+
+        if (previousRootFeedIdentifier != rootFeedIdentifier) {
```

Comment:
Shouldn't the search URL be reset at **every** navigation? (just before a "search" URL is discovered (or not) in the feed) I don't think we should preserve the search URL if we are still on the same "domain" (OPDS root identifier). The "search" functionality is totally dependent on each OPDS feed declaring its available services / URL endpoints (hypermedia model).

Reply:
Please let me know what you think, maybe I missed something. Thank you!